poetry/repositories/pypi_repository.py

```diff
@@ -432,14 +432,17 @@ def _get_info_from_urls(self, urls: Dict[str, List[str]]) -> "PackageInfo":
     def _get_info_from_wheel(self, url: str) -> "PackageInfo":
         from poetry.inspection.info import PackageInfo
 
+        pathsplits = urllib.parse.urlparse(url).path.rsplit("/")
+        wheelpath = pathsplits[-1]
+        if not wheelpath:
+            wheelpath = pathsplits[-2]
+
         self._log(
-            "Downloading wheel: {}".format(
-                urllib.parse.urlparse(url).path.rsplit("/")[-1]
-            ),
+            "Downloading wheel: {}".format(wheelpath),
             level="debug",
         )
 
-        filename = os.path.basename(urllib.parse.urlparse(url).path.rsplit("/")[-1])
+        filename = os.path.basename(wheelpath)
```

Comment:
The basename call here is useless as best I can tell, but it's even more redundant using pathlib. Let's factor it out completely (e.g. remove filename entirely).